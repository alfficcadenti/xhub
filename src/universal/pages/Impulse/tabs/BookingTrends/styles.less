@import '../../../../mixins.less';

.bookings-container-box {
  width: 90%;

  .reset-div {
    display: flex;
    justify-content: flex-end;
    margin-right: @spacing-4x;
    .reset-btn {
      padding: 5px;
      margin-bottom: @spacing-2x;
    }
  }

  .custom-tooltip {
    background-color: rgba(150, 150, 151, .7);
    border-radius: @spacing-2x;
    vertical-align: center;
    text-align: left;

    .label {
      color: @text-color;
    }
  }
}
