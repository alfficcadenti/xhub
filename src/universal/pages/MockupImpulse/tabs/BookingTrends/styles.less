@import '../../../../mixins.less';

.bookings-container-box {
  width: 90%;
}
