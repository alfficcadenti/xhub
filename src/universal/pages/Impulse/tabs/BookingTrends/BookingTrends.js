import React from 'react';
import {Area, AreaChart, CartesianGrid, Legend, ResponsiveContainer, Tooltip, XAxis, YAxis} from 'recharts';
import './styles.less';
import moment from 'moment';

const BOOKING_CHART_COLOR = '#1478F7';
const PREDICTION_CHART_COLOR = '#c9405b';
const PREDICTION_COUNT = 'Prediction Counts';
const BOOKING_COUNT = 'Booking Counts';
const AreaChartType = 'Area';
const IMPULSE_CHART_TYPE = [
    {
        name: BOOKING_COUNT,
        color: BOOKING_CHART_COLOR,
        chartType: AreaChartType,
        key: 'bookingChart'
    },
    {
        name: PREDICTION_COUNT,
        color: PREDICTION_CHART_COLOR,
        chartType: AreaChartType,
        key: 'predictionChart'
    }
];

const formatDateTimeLocal = (date) => moment(date).format('MM/DD HH:mm');

const CustomTooltip = ({active, payload}) => {
    const TIMEZONE = moment().tz(moment.tz.guess()).format('z');
    if (active && payload && payload[0] && payload[0].payload) {
        return (<div className="custom-tooltip">
            <span className="label">{`${formatDateTimeLocal(payload[0].payload.time)} ${TIMEZONE}`}</span>
            {payload.map((item) => (
                <div>
                    <span className="label">{`${item.dataKey} : ${item.value}`}</span>
                </div>
            ))}
        </div>);
    }
    return null;
};

const BookingTrends = ({data = []}) => {
    const getGradient = ({key, color}) => {
        const id = key === 'bookingChart' ? `color${key}` : '';
        return (<linearGradient key={`${key}Gradient`} id={id} x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor={color} stopOpacity={0.8}/>
            <stop offset="95%" stopColor={color} stopOpacity={0}/>
        </linearGradient>);
    };

    const renderChart = ({key, color, name}) => {
        const fill = `url(#color${key})`;
        return (<Area type="monotone" dataKey={name} stroke={color} fillOpacity={1} fill={fill}/>);
    };

    return (
        <div className="bookings-container-box">
            <div className="header-container">
                <h2 className="total-bookings-title">
                    {'Impulse'}
                </h2>
            </div>
            <ResponsiveContainer width="100%" height="70%">
                <AreaChart data={data} margin={{top: 10, right: 30, left: 0, bottom: 0}}>
                    <defs>
                        {IMPULSE_CHART_TYPE.map(getGradient)}
                    </defs>
                    <XAxis dataKey="time" tick={{fontSize: 10}} tickFormatter={formatDateTimeLocal}/>
                    <YAxis tick={{fontSize: 10}}/>
                    <CartesianGrid strokeDasharray="3 3"/>
                    <Tooltip content={<CustomTooltip/>}/>
                    {IMPULSE_CHART_TYPE.map(renderChart)}
                    <Legend/>
                </AreaChart>
            </ResponsiveContainer>
        </div>
    );
};

export default React.memo(BookingTrends);
