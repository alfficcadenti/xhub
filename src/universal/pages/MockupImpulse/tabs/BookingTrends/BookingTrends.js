import React from 'react';
import {Area, AreaChart, CartesianGrid, Legend, ResponsiveContainer, Tooltip, XAxis, YAxis} from 'recharts';
import './styles.less';

const BOOKING_CHART_COLOR = '#1478F7';
const PREDICTION_CHART_COLOR = '#c9405b';
const PREDICTION_COUNT = 'Prediction Counts';
const BOOKING_COUNT = 'Booking Counts';
const AreaChartType = 'Area';
const IMPULSE_CHART_TYPE = [
    {
        name: BOOKING_COUNT,
        color: BOOKING_CHART_COLOR,
        chartType: AreaChartType,
        key: 'bookingChart'
    },
    {
        name: PREDICTION_COUNT,
        color: PREDICTION_CHART_COLOR,
        chartType: AreaChartType,
        key: 'predictionChart'
    }
];


const BookingTrends = ({data = []}) => {
    const getGradient = ({key, color}) => {
        const id = key === 'bookingChart' ? `color${key}` : '';
        return (<linearGradient key={`${key}Gradient`} id={id} x1="0" y1="0" x2="0" y2="1">
            <stop offset="5%" stopColor={color} stopOpacity={0.8}/>
            <stop offset="95%" stopColor={color} stopOpacity={0}/>
        </linearGradient>);
    };

    const renderChart = ({key, color, name}) => {
        const fill = `url(#color${key})`;
        return (<Area type="monotone" dataKey={name} stroke={color} fillOpacity={1} fill={fill}/>);
    };

    return (
        <div className="bookings-container-box">
            <h2 className="total-bookings-title">
                {'Impulse'}
            </h2>
            <ResponsiveContainer width="100%" height="70%">
                <AreaChart data={data} margin={{top: 10, right: 30, left: 0, bottom: 0}}>
                    <defs>
                        {IMPULSE_CHART_TYPE.map((item) =>
                            getGradient(item)
                        )}
                    </defs>
                    <XAxis dataKey="time" tick={{fontSize: 10}}/>
                    <YAxis tick={{fontSize: 10}}/>
                    <CartesianGrid strokeDasharray="3 3"/>
                    <Tooltip/>
                    {IMPULSE_CHART_TYPE.map((item) =>
                        renderChart(item)
                    )}
                    <Legend/>
                </AreaChart>
            </ResponsiveContainer>
        </div>
    );
};

export default React.memo(BookingTrends);
