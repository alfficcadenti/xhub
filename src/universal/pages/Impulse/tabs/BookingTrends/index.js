import BookingTrends from './sections/BookingChart/BookingChart';
import IncidentDetails from './sections/IncidentTable/IncidentDetails';
export {
    BookingTrends,
    IncidentDetails
};
