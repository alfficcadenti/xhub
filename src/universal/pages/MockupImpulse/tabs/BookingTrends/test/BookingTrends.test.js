import BookingTrends from '../BookingTrends';
import {render, shallow} from 'enzyme';
import React from 'react';
import chai, {expect} from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';
import fakeData from './data.test.json';
chai.use(chaiJestSnapshot);
describe('Booking Trend component testing', () => {
    let wrapper;

    beforeEach(() => {
        chaiJestSnapshot.setFilename(`${__filename}.snap`);
        wrapper = shallow(
            <BookingTrends data ={fakeData} />
        );
    });
    afterEach(() => {
        wrapper.unmount();
    });


    it('renders successfully', () => {
        expect(wrapper).to.have.length(1);
    });

    it('matches the snapshot', () => {
        chaiJestSnapshot.setTestName('matches the snapshot');
        const wrapper1 = render(<BookingTrends data={fakeData}/>);
        expect(wrapper1).to.matchSnapshot();
    });
});