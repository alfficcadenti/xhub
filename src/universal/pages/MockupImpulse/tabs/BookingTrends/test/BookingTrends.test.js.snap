// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "echarts-for-react ",
          "style": "height:300px",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
